### ISSUE TITLE
Users unable to creaet a new Che workspace in OpenShift.io

### ISSUE THREAD

**Issue by ldimaggi** [openshiftio/openshift.io#3103]
Steps to recreate:
* Create a new space, new app/quickstart
* Attempt to create a new Che workspace
* The workspace fails to start - hangs on this screen:

![screenshot from 2018-04-12 12-09-04](https://user-images.githubusercontent.com/642621/38690136-5a87cb66-3e4b-11e8-9621-936edf03ba51.png)

No errors are seen in the Che6 log:
```
{"@timestamp":"2018-04-12T16:07:54.886+00:00","@version":1,"message":"Workspace
 'ldimaggi@redhat.com/april12-nyvn3' with id 'workspaceuimpi9knqfw8jkob' created by user
 'ldimaggi@redhat.com'","logger_name":"org.eclipse.che.api.workspace.server.WorkspaceManager"
,"thread_name":"http-nio-8080-exec-13","level":"INFO","level_value":20000,"req_id"
:"6ca0e5a5-077a-4134-aa0a-e524b7ea3db9","identity_id":"20ddc23a-bb62-4834-9130-9af2f54e85b1"}
```

**Comment by ibuziuk** [comment]
Related to https://github.com/redhat-developer/rh-che/issues/629

**Comment by ibuziuk** [comment]
> Related to the number of concurrent users in Che?
might be @rhopp is going to perform some load tesing AFAIK

**Comment by eivantsov** [comment]
@ibuziuk is it still Che starter that is in charge of staring workspaces? According to server log, the workspace has been created but never started. Auto start of workspaces is set to false as far as I understand. Hence, when IDE link is opened in a new tab, there's this message and the IDE does not auto start the workspace.

**Comment by ibuziuk** [comment]
@eivantsov yeah, che-starter is in charge of starting workpaces. It seems that the issue occurs under some load when dozen of users tries to start a workspace. Assuming that workspace startup is implemented asynchronously it might be a matter of configuring async thread pool executor on che-starter side.

However, if you take a look at description of rh-che issue https://github.com/redhat-developer/rh-che/issues/629 the weird part there is `The issue is getting problematic when the user start a second workspace:` Che treats first workspace as "Started / Starting" even thought it is stopped - https://github.com/redhat-developer/rh-che/issues/629#issuecomment-380449398

WDYT about this one ? ^

Also, integration tests in che-starter caught some unexpected problems with workspace status flapping between running & starting - https://github.com/redhat-developer/rh-che/issues/629#issuecomment-380496583

one more thing is similar problem with dashboard https://github.com/eclipse/che/issues/9381
(not sure if it related, but the context is quite the same)

**Comment by ldimaggi** [comment]
Are the parameters to force the error: 10 users - regardless of whether their accounts are configured on the same or different clusters - if they all attempt to start a workspace in the same ~2 minute window?

**Comment by ldimaggi** [comment]
That's fine - if you believe that the same issue is described in both issues.
